texttospeech/api/TextToSpeech/TextToSpeech.cs

```diff
@@ -0,0 +1,288 @@
+﻿/*
+ * Copyright (c) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using CommandLine;
+using System;
+using System.IO;
+using Google.Cloud.TextToSpeech.V1;
+
+namespace GoogleCloudSamples
+{
+
+    [Verb("list", HelpText = "List available voices.")]
+    class ListArgs
+    {}
+
+    [Verb("synthesize", HelpText = "Synthesize input to audio")]
+    class SynthesizeArgs
+    {
+        [Value(0, HelpText = "The text to translate.",
+            Required = true)]
+        public string Text { get; set; }
+
+        [Option('f', HelpText = "Source formatting")]
+        public string SourceFormat { get; set; }
+
+    }
+
+    [Verb("synthesize-file", HelpText = "Synthesize a file to audio")]
+    class SynthesizeFileArgs
+    {
+        [Value(0, HelpText = "The file to synthesize",
+            Required = true)]
+        public string Text { get; set; }
+
+        [Option('f', HelpText = "Source formatting")]
+        public string SourceFormat { get; set; }
+    }
+
+    public class TextToSpeech
+    {
+        public static void Main(string[] args)
+        {
+            Console.OutputEncoding = System.Text.Encoding.Unicode;
+            Parser.Default.ParseArguments<ListArgs, SynthesizeArgs,
+                SynthesizeFileArgs>(args).MapResult(
+                (ListArgs largs) => ListVoices(largs),
+                (SynthesizeArgs sargs) => Synthesize(sargs),
+                (SynthesizeFileArgs sfargs) => SynthesizeFile(sfargs),
+                errs => 1);
+        }
+
+        private static int ListVoices(ListArgs args)
+        {
+            ListVoices();
+            return 0;
+        }
+
+        // [START tts_list_voices]
+        /// <summary>
+        /// Lists all the voices available for speech synthesis.
+        /// </summary>
+        public static void ListVoices()
+        {
+            TextToSpeechClient client = TextToSpeechClient.Create();
+
+            // Performs the list voices request
+            var response = client.ListVoices(new ListVoicesRequest
+            {
+                // Uncomment the following block and specify a language
+                // code to get results for that language only.
+                //LanguageCode = "[LANGUAGE_CODE]"
```

Comment:
nit `// LanguageCode =`

Reply:
So, I did something *a little* different than your suggestions but more like the Translate API example that you sent. I got rid of the private ListVoices method, made the public method return an int, and set it so that you can call the method either passing in a language code (or not).